### PR TITLE
Add cmake option to enable/disable installation Issue #1029

### PR DESCRIPTION
**Pull request by ptc-tgamper**
draco_options.cmake - add DRACO_INSTALL option;
draco_install.cmake - check DRACO_INSTALL before adding install rules

### CODE REVIEW

File: cmake/draco_install.cmake [hunk 1]
```diff
@@ -23,99 +23,101 @@ include(GNUInstallDirs)
...
 macro(draco_setup_install_target)
-  set(bin_path "${CMAKE_INSTALL_BINDIR}")
-  set(data_path "${CMAKE_INSTALL_DATAROOTDIR}")
-  set(includes_path "${CMAKE_INSTALL_INCLUDEDIR}")
-  set(libs_path "${CMAKE_INSTALL_LIBDIR}")
-
-  foreach(file ${draco_sources})
-    if(file MATCHES "h$")
-      list(APPEND draco_api_includes ${file})
+  if (DRACO_INSTALL)
```
Comment (ptc-tgamper) [inline cmake/draco_install.cmake:26]: Because this is a macro we cannot simpliy do an early return().

Comment (jzern) [reply cmake/draco_install.cmake:26]: This looks like it should work. A couple of comments:

1. What is the reason for the change though? Were you running into an issue in the cmake invocation?
2. Change this line to `if(DRACO_INSTALL)` (no space after `if`) to match the rest of the code.

Comment (jzern) [reply cmake/draco_install.cmake:26]: > This looks like it should work. A couple of comments:
> 
> 1. What is the reason for the change though? Were you running into an issue in the cmake invocation?

I missed the reference to #1029. The second point should still be addressed.

> 2. Change this line to `if(DRACO_INSTALL)` (no space after `if`) to match the rest of the code.

Comment (ptc-tgamper) [reply cmake/draco_install.cmake:26]: The reason for the change is a project where we pull in draco via cmake FetchContent as a dependency. We found that we didn't have any control over the installation of draco files, they always got installed if we invoked cmake install, be it for a single target or the ALL target. But draco is only linked as a private dependency, we don't want to install it, we don't want its files ending up in the installation directory.

**Review by ondys — APPROVED** [review]

**Pull request merged**